stackit/internal/services/cdn/customdomain/resource.go

```diff
@@ -260,30 +369,130 @@ func (r *customDomainResource) ImportState(ctx context.Context, req resource.Imp
 	tflog.Info(ctx, "CDN custom domain state imported")
 }
 
-func mapCustomDomainFields(customDomain *cdn.CustomDomain, model *CustomDomainModel) error {
-	if customDomain == nil {
+func NormalizeCertificate(certInput cdn.GetCustomDomainResponseGetCertificateAttributeType) (Certificate, error) {
```

Comment:
The change for making this function public can be reverted, because it's only required within the customdomain package
```suggestion
func normalizeCertificate(certInput cdn.GetCustomDomainResponseGetCertificateAttributeType) (Certificate, error) {
```